src/procedures/TransferOwnership.ts

```diff
@@ -0,0 +1,81 @@
+import { Procedure } from './Procedure';
+import {
+  ProcedureType,
+  PolyTransactionTag,
+  ErrorCode,
+  TransferOwnershipProcedureArgs,
+} from '../types';
+import { PolymathError } from '../PolymathError';
+import { SecurityToken } from '../entities';
+import { Factories } from '../Context';
+import { isValidAddress } from '../utils';
+
+/**
+ * @hidden
+ */
+export const createTransferOwnershipResolver = (
+  factories: Factories,
+  symbol: string
+) => async () => {
+  return factories.securityTokenFactory.refresh(SecurityToken.generateId({ symbol }));
+};
+
+/**
+ * Procedure that will transfer the ownership of a Security Token to a different wallet
+ */
+export class TransferOwnership extends Procedure<TransferOwnershipProcedureArgs> {
+  public type = ProcedureType.TransferOwnership;
+
+  /**
+   * - Transfer ownership of a Security Token to another wallet
+   *
+   * Note this procedure will fail if:
+   * - Trying to call the procedure from a wallet that is not the current owner of the Security Token
+   * - Transferring the ownership to the current owner, as there is no change
+   */
+  public async prepareTransactions() {
+    const { newOwner, symbol } = this.args;
+    const { contractWrappers, currentWallet, factories } = this.context;
+
+    let securityToken;
+
+    try {
+      securityToken = await contractWrappers.tokenFactory.getSecurityTokenInstanceFromTicker(
+        symbol
+      );
+    } catch (err) {
+      throw new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: `There is no Security Token with symbol ${symbol}`,
+      });
+    }
+
+    const owner = await securityToken.owner();
+
+    if ((await currentWallet.address()) !== owner) {
```

Comment:
Try not to await inside conditional expressions, instead store the result in a variable and check for that. Also, you can get the owner and the current wallet in parallel